cli/parse_args.ts

```diff
@@ -18,7 +18,9 @@
  */
 import { assert } from "../assert/assert.ts";
 
-/** Combines recursively all intersection types and returns a new single type. */
+/** Combines recursively all intersection types and returns a new single type.
+ * @internal
```

Comment:
This is to suppress the "type is used in public type, but not exported" error in `deno doc --lint`

Reply:
Hm... I thought exporting such types was actually a good thing. In my previous PRs for #3764, I exported them.

Reply:
Depends if someone might need to depend on the type or not. If it's just some internal alias then it might not be useful to export. Perhaps I should revert the `@internal` changes and someone with the proper knowledge can go through to figure out what to export?

Reply:
Yeah, that's a good idea. I can take care of those.

Reply:
Done.